python/cudf/cudf/_lib/column.pyx

```diff
@@ -203,10 +207,27 @@ cdef class Column:
             "The value for mask is smaller than expected, got {}  bytes, "
             "expected " + str(required_num_bytes) + " bytes."
         )
+
+        # Because hasattr will trigger invocation of
+        # `__cuda_array_interface__` which could
+        # be expensive in CopyOnWriteBuffer case.
+        value_cai = inspect.getattr_static(
+            value,
+            "__cuda_array_interface__",
+            None
+        )
+
         if value is None:
             mask = None
-        elif hasattr(value, "__cuda_array_interface__"):
-            if value.__cuda_array_interface__["typestr"] not in ("|i1", "|u1"):
+        elif type(value_cai) is property:
+            if isinstance(value, CopyOnWriteBuffer):
+                value = SimpleNamespace(
```

Comment:
Changed all `SimpleNamespace` to be replaced with `_cuda_array_interface_readonly` calls which will go through `cuda_array_interface_wrapper`. Haven't moved `cuda_array_interface_wrapper` from `buffer.py` to `buffer/utils.py` in this PR because we will be running into circular import issues.